endpoints/events/event.go

```diff
@@ -107,10 +107,16 @@ func (e *eventEndpoint) Handle(w http.ResponseWriter, r *http.Request, _ httprou
 		return
 	}
 
-	// account does not support events
-	if !account.EventsEnabled {
+	// account does not support events and events are not enabled by default for accounts
+	if account.EventsEnabled != nil {
+		if !*account.EventsEnabled {
+			w.WriteHeader(http.StatusUnauthorized)
+			w.Write([]byte(fmt.Sprintf("Account '%s' doesn't support events", eventRequest.AccountID)))
+			return
+		}
+	} else if e.Cfg.AccountDefaults.EventsEnabled == nil || !*e.Cfg.AccountDefaults.EventsEnabled {
```

Comment:
Hi @shahinrahbariasl sorry for not coming back to you sooner. Your code looks pretty good but, after discussing this feature behind the scenes we notice that `"events_enabled"`, which is deprecated, is still in use in some incoming requests so we'd like to still use its value whenever  `"events.enabled"` is not found. In other words, we should prioritize `account.Events.Enabled` over `account.EventsEnabled`. Can we please code this logic, put it into its own function and write unit tests for it? Please set the resulting boolean value into `account.Events.Enabled` and set `account.EventsEnabled` to `nil`.
| `"events_enabled"` | `"events.enabled"` | output |
| ------ | ------ | ------ |
| `false` | `false` | `false` |
| `false` | `true` | `true` |
| `false` | not set | `false` |
| `true` | `false` | `false` |
| `true` | `true` | `true` |
| `true` | not set | `true` |
| not set | `false` | `false` |
| not set | `true` | `true` |
| not set | not set | `true` |

We should use the same logic shown above for `e.Cfg.AccountDefaults.EventsEnabled` and `e.Cfg.AccountDefaults.Events.Enabled` before the `MarshalAccountDefaults()` function call in line 691 of `config/config.go` but, in this instance, set the resulting value to the deprecated field `e.Cfg.AccountDefaults.EventsEnabled` while setting the new field `e.Cfg.AccountDefaults.Events.Enabled` to `nil`.
```
673 // New uses viper to get our server configurations.
674 func New(v *viper.Viper, bidderInfos BidderInfos, normalizeBidderName func(string) (openrtb_ext.BidderName, bool)) (*Configuration, error) {
675     var c Configuration
676     if err := v.Unmarshal(&c); err != nil {
677         return nil, fmt.Errorf("viper failed to unmarshal app config: %v", err)
678     }
679
680     if err := c.RequestValidation.Parse(); err != nil {
681         return nil, err
682     }
683
684     if err := isValidCookieSize(c.HostCookie.MaxCookieSizeBytes); err != nil {
685         glog.Fatal(fmt.Printf("Max cookie size %d cannot be less than %d \n", c.HostCookie.MaxCookieSizeBytes, MIN_COOKIE_SIZE_BYTES))
686         return nil, err
687     }
688
689     // Update account defaults and generate base json for patch
690     c.AccountDefaults.CacheTTL = c.CacheURL.DefaultTTLs // comment this out to set explicitly in config
691     if err := c.MarshalAccountDefaults(); err != nil {
692         return nil, err
693     }
694
config/config.go
```
In summary we want to maintain the following order of precedence:
1. account config new field `account.Events.Enabled`
2. account config deprecated field `account.EventsEnabled`
3. host config new field `e.Cfg.AccountDefaults.Events.Enabled`
4. host config deprecated field `e.Cfg.AccountDefaults.EventsEnabled`
Does this make sense?